### PR TITLE
Feature - payment upon trigger

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

File: Model/Adminhtml/Config/PaymentPlans.php [hunk 1]
```diff
@@ -139,4 +156,21 @@ public function beforeSave()
 
         return parent::beforeSave();
     }
+
+    /**
+     * @param $plans
+     * @return array
+     */
+    private function correctMinMaxLimit($plans):array
```
Comment (syjust) [inline Model/Adminhtml/Config/PaymentPlans.php:164]: Can you add a description and type parameter properly (I would like to know `$plans` is array of what 😉 )

This method is a little bit dirty (I'am not fan of "correcting" or "fixing" value previously built (maybe we hab to rework this information before this step)...

I let you decide if we have to do this now or if this is the best you can do according our short delay to deliver this feature (discussion is open if you want to talk about that now)

File: Model/Adminhtml/Config/TriggerIsAllowed.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var PaymentPlansHelper
+     */
+    private $paymentPlansHelper;
+
+    public function __construct(
+        Logger $logger,
```
Comment (syjust) [inline Model/Adminhtml/Config/TriggerIsAllowed.php:23]: logger is not used

File: Model/Adminhtml/Config/TriggerIsAllowed.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function toOptionArray()
+    {
+        $isAllowedValue = 0;
+        $isAllowedLabel = self::NOT_ALLOWED_LABEL;
+        if($this->paymentPlansHelper->paymentTriggerIsAllowed()){
+            $isAllowedValue = 1;
+            $isAllowedLabel = self::ALLOWED_LABEL;
+        }
+        return [['value' => $isAllowedValue, 'label' => __($isAllowedLabel)]];
```
Comment (syjust) [inline Model/Adminhtml/Config/TriggerIsAllowed.php:50]: warning: translate a variable don't will add translation key in our translations files !

File: Model/Adminhtml/Config/TriggerTypology.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Options getter
+     *
+     * @return array
+     */
+    public function toOptionArray(): array
+    {
+        $optionArray = [];
+        foreach (self::TRIGGER_LIST as $trigger) {
+            $optionArray[]=['value' => $trigger['value'], 'label' => __($trigger['label'])];
```
Comment (syjust) [inline Model/Adminhtml/Config/TriggerTypology.php:34]: warning: translate a variable don't will add translation key in our translations files !

File: etc/adminhtml/system.xml [hunk 5]
```diff
@@ -39,6 +39,7 @@
                        showInStore="1">
                     <label>Enabled</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <frontend_model>Alma\MonthlyPayments\Block\Adminhtml\Form\Field\ModuleVersion</frontend_model>
```
Comment (syjust) [inline etc/adminhtml/system.xml:42]: 😍

File: i18n/en_US.csv [hunk 6]
```diff
@@ -110,3 +110,10 @@
 "Customize Alma in your checkout","Customize Alma in your checkout"
 "Edit the text displayed when choosing the payment method in your checkout. It will adapt to the languages of your site.","Edit the text displayed when choosing the payment method in your checkout. It will adapt to the languages of your site."
 "Merge all payment methods","Merge all payment methods"
+"Payment upon trigger","Payment upon trigger"
```
Comment (syjust) [inline i18n/en_US.csv:113]: did you add this translations keys manually or is there a magento command to build translations files ?

File: view/frontend/web/template/payment/installments.html [hunk 7]
```diff
@@ -37,37 +37,42 @@
          data-bind="attr: { src: require.toUrl('Alma_MonthlyPayments/images/alma_logo.svg'), alt: 'alma'}">
     <div class="payment-method-content">
         <!-- ko foreach: getRegion('messages') -->
-        <!-- ko template: getTemplate() --><!-- /ko -->
+            <!-- ko template: getTemplate() --><!-- /ko -->
```
Comment (syjust) [inline view/frontend/web/template/payment/installments.html:40]: lint 😛

File: Model/Adminhtml/Config/TriggerIsAllowed.php (hunk 3 above)
Comment (Francois-Gomis) [reply Model/Adminhtml/Config/TriggerIsAllowed.php:50]: it's "yes" or "no"  label they are already translated in default

File: i18n/en_US.csv (hunk 6 above)
Comment (Francois-Gomis) [reply i18n/en_US.csv:113]: It's a manually add

**Pull request merged**